### PR TITLE
Feature/ot221 66

### PR DESCRIPTION
**Pull request by rodrigo1995vav**
## Issue ticket link
https://alkemy-labs.atlassian.net/browse/OT221-66

## Summary
-Added post method to send an image file and text fields to the AWS server and to the activities table respectively
-Added middleware to check fields and file 
-Added get method to retrieve image from AWS server
-Multer is used to upload files.
-If the client sends a file, Multer will check his proprieties, but won't abort the post since the client may or may not send a file. In the event that a file has to strictly be sent to the server, apply fileValidation as middleware in the post method

## Evidence
-In the event that name field does not exist, is empty, or is too short. Post req is canceled and it throws errors 
2022-07-07 15_04_58-Postman

-In the event that the image file doesn't match specifications or/and it is too big. Post req is canceled, throws errors and the file gets deleted.
2022-07-08 20_42_43-Postman

-If everything is "ok", data is stored in S3 AWS and in the database
2022-07-07 15_06_06-Postman
2022-07-07 15_11_08-DBeaver 22 0 3 - activities

-When requesting the uploaded file to the AWS server, everything goes right
2022-07-07 15_06_56-Postman

### CODE REVIEW

File: src/middlewares/field-validator.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
 const { validationResult } = require("express-validator");
+const { deleteTempFile } = require("../services/fileServices");
 
 const validateFields = (req, res, next) => {
 
     const errors = validationResult(req);
-    if(!errors.isEmpty()){
+    //Since the validation of the fields is made outside multer, multer will upload the file to the server first, so in the event that the fields are invalid the file must be deleted.
+    if(!errors.isEmpty()){ 
+        if (req.file){deleteTempFile(req.file.path)}
```
Comment (GauntStrelok) [inline src/middlewares/field-validator.js:9]: Esto no borra el archivo de s3

File: src/middlewares/validateExtensionsMulter.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const validateExtensionMulter = (req, file, cb) => { //
+    const { mimetype } = file;
+    const fileType = mimetype.split("/")[0];
+    if (fileType === "image") {
+      cb(null, true);
+    } else {
+      cb(null, false);
+      return cb(new Error("Only .png, .jpg and .jpeg format allowed!"));
```
Comment (GauntStrelok) [inline src/middlewares/validateExtensionsMulter.js:8]: Estaria bueno que el error fuera standarizado, y no devuelva de diferente forma con cada error, aca devolves un string, cuando en las otras validaciones devolves un objeto con un errors adentro que tiene un array de objetos...

Comment (rodrigo1995vav) [reply src/middlewares/validateExtensionsMulter.js]: Ahí estandarice los errores

File: src/middlewares/file-validator.js [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
-    if(fileType !== 'image'){
-      return res.status(400).json({
-        ok: false,
-        err: 'Incorrect file format.'
-      });
-    }
+  // it extracts the filetype of the request.files.file object
+  if (fileType !== "image") {
+    deleteTempFile(req.file.path)
+    return res.status(400).json({error:{ message: "El archivo debe ser una imagen." }});
```
Comment (GauntStrelok) [inline src/middlewares/file-validator.js:24]: Mira que en el otro caso el errors es un array de objetos, aca es solo un objeto

Comment (rodrigo1995vav) [reply src/middlewares/file-validator.js]: Listo ya lo resolví, adjunto imagen en la evidencia

**Pull request merged**